code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -929,12 +992,35 @@ func (sc *StorageSmartContract) extendAllocation(
 		}
 	}
 
-	// add more tokens to related challenge pool, or move some tokens back
-	var remainingDuration = alloc.Expiration - txn.CreationDate
-	err = sc.adjustChallengePool(alloc, originalRemainingDuration, remainingDuration, originalTerms, conf.TimeUnit, balances)
-	if err != nil {
-		return common.NewErrorf("allocation_extending_failed", "%v", err)
+	// No challenge pool for enterprise blobber
+	actErr = chainstate.WithActivation(balances, "electra", func() error {
+		var remainingDuration = alloc.Expiration - txn.CreationDate
+		err = sc.adjustChallengePool(alloc, originalRemainingDuration, remainingDuration, originalTerms, conf.TimeUnit, balances)
+		if err != nil {
+			return common.NewErrorf("allocation_extending_failed", "%v", err)
+		}
+		return nil
+	}, func() error {
+		if isEnterprise {
+			var remainingDuration = alloc.Expiration - txn.CreationDate
+			err = sc.adjustChallengePool(alloc, originalRemainingDuration, remainingDuration, originalTerms, conf.TimeUnit, balances)
+			if err != nil {
+				return common.NewErrorf("allocation_extending_failed", "%v", err)
+			}
+		} else {
+			var remainingDuration = alloc.Expiration - txn.CreationDate
+			err = sc.adjustChallengePool(alloc, originalRemainingDuration, remainingDuration, originalTerms, conf.TimeUnit, balances)
+			if err != nil {
+				return common.NewErrorf("allocation_extending_failed", "%v", err)
+			}
```

Comment:
Looks like the both enterprise and none enterprise perform the same logic

Reply:
Ah yeah, missed it. Fixed.